active/0011-jq-in-rule-engine.md

```diff
@@ -146,13 +147,67 @@ only output a single map result, and the map will then be piped to the JQ clause
 
 I am happy with this syntax now.
 
-### Introduce JQ as NIF
+### Introduce JQ as Port or NIF
 
+JQ is written in portable C as a single binary, reading command line argument
+from stdin and outputting results to the stdout. It supports Linux, OS X,
+FreeBSD, Solaris, and Windows for now, so the simplest way is to package the `jq`
+binary along with all of the emqx installation packages, and talk to `jq` using
+the [erlang port](http://erlang.org/doc/tutorial/c_port.html). For someone who
```

Comment:
emqx probably needs multi port programs in order to make things run in parallel?

Reply:
Yes, we need to start multiple ports, each for a rule (added by emqx rule engine).

Reply:
As you know, the current implementation use a fixed number of long running port porgrams.